models/webhook/webhook.go

```diff
@@ -22,6 +22,38 @@ import (
 	"xorm.io/builder"
 )
 
+// MetaSettings represents the metadata settings for webhook
+type MetaSettings struct {
+	PayloadOptimization *PayloadOptimizationConfig `json:"payload_optimization,omitempty"` // Payload optimization configuration
+}
+
+// PayloadOptimizationConfig represents the configuration for webhook payload optimization
+type PayloadOptimizationConfig struct {
+	Files   *PayloadOptimizationItem `json:"files,omitempty"`   // Files optimization config
+	Commits *PayloadOptimizationItem `json:"commits,omitempty"` // Commits optimization config
+}
+
+// PayloadOptimizationItem represents a single optimization item configuration
+type PayloadOptimizationItem struct {
```

Comment:
`Optimization` reads strange. `Optimization` means "improve something while keep the result the same as before"

Here it is likely just a "config" system.